### ISSUE TITLE
SEGV with two ibm onesided tests, tcp btl, and debug Open MPI

### ISSUE THREAD

**Issue by rolfv** [open-mpi/ompi#338]
I have noticed that I am getting a SEGV when I build the post 1.8.4 version of Open MPI when running two of the IBM onesided tests.  This may also be in 1.8.4 - I have not checked.  This is what the failure looks like.
```
[rvandevaart@drossetti-ivy0 onesided]$ mpirun -np 2 --mca btl tcp,self ./c_strided_acc_onelock 
--------------------------------------------------------------------------
mpirun noticed that process rank 0 with PID 18567 on node drossetti-ivy0 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
[rvandevaart@drossetti-ivy0 onesided]$ 
```

I have isolated the problem to a specific line of code, but I am not sure what the problem is.  The error occurs in the MCA_BTL_TCP_ENDPOINT_DUMP macro as shown below (btl_tcp_endpoint.c:356)
```
        } else {
            MCA_BTL_TCP_ENDPOINT_DUMP(10, btl_endpoint, true, "send fragment enqueued [endpoint_send]");
            frag->base.des_flags |= MCA_BTL_DES_SEND_ALWAYS_CALLBACK;
            opal_list_append(&btl_endpoint->endpoint_frags, (opal_list_item_t*)frag);
  ```
The macro makes a call to mca_btl_tcp_endpoint_dump() and within that function things blow up at line 263
```
   OPAL_LIST_FOREACH(item, &btl_endpoint->endpoint_frags, mca_btl_tcp_frag_t) {
```
I assume there is something wrong in terms of initialization but...not sure.  
* Note that you only see this if you configure with --enable-debug. 
* I ran it with valgrind and it complained about invalid memory at &btl_endpoint->endpoint_frags
* If I comment out the MACRO, the error goes away
* I only see this error with these two ibm onesided tests. 
 * ibm/onesided/c_strided_acc_onelock 
 * ibm/onesided/c_accumulate_atomic

**Comment by ggouaillardet** [comment]
@rolfv i fixed this in open-mpi/ompi@8ab605d9c5bc82ef2baf424a5d5f970132d89ca1

snprintf() output was incorrectly used, per the man page
```
The functions snprintf() and vsnprintf() do not write more than size bytes (including the trailing '\0').  If the output was truncated due to  this limit  then  the  return  value  is the number of characters (not including the trailing '\0') which would have been written to the final string if enough space had been available.
```

overflow are now correctly handled.

**Comment by bosilca** [comment]
open-mpi/ompi@8ddd3b3 and open-mpi/ompi@1344097 should also be associated with this fix.
